### PR TITLE
#2068: disable vtAssert test for Intel Classic

### LINKED ISSUE #2068: `TestPreconfig.test_vt_assert_no_mpi` segfaults in `icpx` build
Reproduced on `develop` (9089be58b) using Docker.
```
root@e89cb46df633:/build/vt# ctest -R TestPreconfig.test_vt_assert_no_mpi --output-on-failure
Test project /build/vt
    Start 995: vt:TestPreconfig.test_vt_assert_no_mpi
1/1 Test #995: vt:TestPreconfig.test_vt_assert_no_mpi ...***Exception: SegFault  0.06 sec
Note: Google Test filter = TestPreconfig.test_vt_assert
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestPreconfig
[ RUN      ] TestPreconfig.test_vt_assert


0% tests passed, 1 tests failed out of 1

Label Time Summary:
unit_test    =   0.06 sec*proc (1 test)

Total Test time (real) =   0.14 sec

The following tests FAILED:
	995 - vt:TestPreconfig.test_vt_assert_no_mpi (SEGFAULT)
```

First observed in #2058: see [Azure logs](https://dev.azure.com/DARMA-tasking/DARMA/_build/results?buildId=46390&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=28db5144-7e5d-5c90-2820-8676d630d9d2).

I suspect that this is caused by an `icpx` update (we don't use pinned version in `ci/docker/ubuntu-intel-oneapi-cpp.dockerfile`).
```
root@062055a887d5:/build/vt# icpx --version
Intel(R) oneAPI DPC++/C++ Compiler 2023.0.0 (2023.0.0.20221201)
```

_Originally posted by @cz4rs in https://github.com/DARMA-tasking/vt/issues/2058#issuecomment-1382259950_

### PR DESCRIPTION
**Pull request by cz4rs**
fixes #2068

### CODE REVIEW

File: tests/unit/runtime/test_preconfig.nompi.cc [hunk 1]
```diff
@@ -51,7 +51,7 @@ namespace vt { namespace tests { namespace unit {
 
 struct TestPreconfig : TestHarness { };
 
-#if not vt_check_enabled(production_build)
+#if not vt_check_enabled(production_build) and not defined(__INTEL_COMPILER)
```
Comment (cz4rs) [inline tests/unit/runtime/test_preconfig.nompi.cc:54]: https://www.intel.com/content/www/us/en/developer/articles/technical/use-predefined-macros-for-specific-code-for-intel-dpcpp-compiler-intel-cpp-compiler-intel-cpp-compiler-classic.html

Comment (PhilMiller) [inline tests/unit/runtime/test_preconfig.nompi.cc:54]: Following Kokkos practice, it's maybe preferable to conditionalize inside the test, and call `GTEST_SKIP` in the case where the test won't be run.

**Review by nlslatt — APPROVED** [review]
Looks good

**Review by JacobDomagala — APPROVED** [review]

File: tests/unit/runtime/test_preconfig.nompi.cc (hunk 1 above)
Comment (cz4rs) [reply tests/unit/runtime/test_preconfig.nompi.cc]:
```
2023-03-22T18:05:03.4641692Z         Start 387: vt:TestPreconfig.test_vt_assert_no_mpi
2023-03-22T18:05:03.4745283Z 387/487 Test #387: vt:TestPreconfig.test_vt_assert_no_mpi .............................................***Skipped   0.01 sec
```

**Review by PhilMiller — APPROVED** [review]

**Pull request merged**